beacon-chain/stategen/replay.go

```diff
@@ -51,7 +51,17 @@ func (s *State) loadBlocks(ctx context.Context, startSlot uint64, endSlot uint64
 	}
 
 	// The last retrieved block root has to match input end block root.
+	// Covers the edge case if there's multiple blocks on the same end slot,
+	// the end root may not be the last index in `blockRoots`.
 	length := len(blocks)
```

Comment:
Please check that length is at least 3 or you are going to get a panic on line 57

Reply:
Ahh. Nice catch